api/src/main/java/org/motechproject/nms/api/web/UserController.java

```diff
@@ -31,47 +39,123 @@
 
     @Autowired
     private LanguageService languageService;
+
     @Autowired
     private KilkariService kilkariService;
 
+    @Autowired
+    private FrontLineWorkerService frontLineWorkerService;
+
+    @Autowired
+    private ServiceUsageService serviceUsageService;
+
+    @Autowired
+    private ServiceUsageCapService serviceUsageCapService;
+
     @RequestMapping("/{serviceName}/user")
     @ResponseBody
     public ResponseUser user(@PathVariable String serviceName, @RequestParam String callingNumber,
                              @RequestParam String operator, @RequestParam String circle, @RequestParam String callId)
             throws NotFoundException {
-        List<Language> languages = languageService.getCircleLanguages(circle);
         StringBuilder failureReasons = validate(callingNumber, operator, circle, callId);
+        if (failureReasons.length() > 0) {
+            throw new IllegalArgumentException(failureReasons.toString());
+        }
 
         ResponseUser user = null;
 
-        if (MOBILE_ACADEMY.equals(serviceName)) {
-            user = new MobileAcademyUser();
-        } else if (MOBILE_KUNJI.equals(serviceName)) {
-            user = new MobileKunjiUser();
-        } else if (KILKARI.equals(serviceName)) {
-            user = new KilkariResponseUser();
-            Subscriber subscriber = kilkariService.getSubscriber(callingNumber);
-            if (subscriber == null) {
-                throw new NotFoundException(String.format(NOT_FOUND, "callingNumber"));
-            }
-            Set<Subscription> subscriptions = subscriber.getSubscriptions();
-            Set<String> packs = new HashSet<>();
-            for (Subscription subscription : subscriptions) {
-                packs.add(subscription.getSubscriptionPack().getName());
-            }
-            ((KilkariResponseUser) user).setSubscriptionPackList(packs);
-        } else {
+        /*
+        Make sure the url the user hit corresponds to a service we are expecting
+         */
+        if (!(MOBILE_ACADEMY.equals(serviceName) || MOBILE_KUNJI.equals(serviceName)) && !(KILKARI.equals(serviceName))) {
             failureReasons.append(String.format(INVALID, "serviceName"));
         }
 
+        /*
+        Handle the FLW services
+         */
+        if (MOBILE_ACADEMY.equals(serviceName) || MOBILE_KUNJI.equals(serviceName)) {
+            user = getFrontLineWorkerResponseUser(serviceName, callingNumber);
+        }
+
+        /*
+        Kilkari in the house!
+         */
+        if (KILKARI.equals(serviceName)) {
+            user = getKilkariResponseUser(callingNumber);
+        }
+
```

Comment:
Why not:
if (MA || MK) {
...
} else if (KK) {
...
} else {
invalid service name
}